Counterexamples/VitaliSetNotMeasurable.lean

```diff
@@ -0,0 +1,318 @@
+/-
+Copyright (c) 2024-present Ching-Tsun Chou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ching-Tsun Chou <chingtsun.chou@gmail.com>
+-/
+
+import Mathlib.MeasureTheory.Measure.Lebesgue.Basic
+import Mathlib.MeasureTheory.Measure.NullMeasurable
+
+/-!
+# Vitali set and its non-measurability
+
+This file defines the Vitali set and proves that it is not measurable.
+The proof is essentially the one in Wikipedia:
+
+## References
+
+* <https://en.wikipedia.org/wiki/Vitali_set>
+-/
+
+open Set Filter Real MeasureTheory
+
+noncomputable section
+
+/-- We first enumerate all measure theoretic results that we will need.  We do so
+because we want to make clear exactly which properties of measurability and the volume
+measure are used in the proof of the non-measurability of the Vitali set. -/
+
+example {a b : ℝ} : volume (Icc a b) = ENNReal.ofReal (b - a) :=
+  volume_Icc
+
+lemma volume_mono {s t : Set ℝ} (h : s ⊆ t) : volume s ≤ volume t := by
+  exact OuterMeasureClass.measure_mono volume h
+
+lemma measurable_nullmeasurable {s : Set ℝ} (h : MeasurableSet s) : NullMeasurableSet s volume :=
+  h.nullMeasurableSet
+
+lemma measurable_null_nullmeasurable {s t : Set ℝ}
+    (hm : NullMeasurableSet s) (hn : volume t = 0) : NullMeasurableSet (s ∪ t) :=
+  NullMeasurableSet.union_null hm hn
+
+lemma nullmeasurable_measurable_null {s : Set ℝ} (h : NullMeasurableSet s volume) :
+    ∃ t ⊆ s, MeasurableSet t ∧ volume t = volume s ∧ volume (s \ t) = 0 := by
+  have ⟨t, t_sub_s, t_m, t_eq_s⟩ := h.exists_measurable_subset_ae_eq
+  use t, t_sub_s, t_m
+  constructor
+  · exact measure_congr t_eq_s
+  · exact ae_le_set.mp t_eq_s.symm.le
+
+lemma shift_volume (s : Set ℝ) (c : ℝ) : volume ((fun x ↦ x + c) '' s) = volume s := by
```

Comment:
You can write this as
```suggestion
lemma shift_volume (s : Set ℝ) (c : ℝ) : volume ((· + c) '' s) = volume s := by
```
Still, I don't think we need to have this theorem. It can just be inlined, or even replaced by a `simp` call.